src/UID2.Client/UID2Encryption.cs

```diff
@@ -172,9 +178,24 @@ private static DecryptionResponse DecryptV2(byte[] encryptedId, KeyContainer key
                 return new DecryptionResponse(DecryptionStatus.UserOptedOut, null, established, siteId, siteKey.SiteId, privacyBits.IsClientSideGenerated);
             }
 
+            if (!IsValidDomainNameForSite(privacyBits, siteId, domainName, keys))
+            {
+                return new DecryptionResponse(DecryptionStatus.DomainNameCheckFailed, null, established, siteId, siteKey.SiteId, privacyBits.IsClientSideGenerated);
+            }
+
             return new DecryptionResponse(DecryptionStatus.Success, idString, established, siteId, siteKey.SiteId, privacyBits.IsClientSideGenerated);
         }
 
+        private static bool IsValidDomainNameForSite(PrivacyBits privacyBits, int siteId, string domainName, KeyContainer keys)
```

Comment:
Not sure about this name, any suggestions?

Reply:
still IsDomainNameAllowedForSite?

Reply:
Renamed as above